pkg/webhook/deployment.go

```diff
@@ -101,6 +101,34 @@ func getSpecDeployment(webhooksSecretName, namespace string) (*appsv1.Deployment
 							Image:           images.GetWebhookServerImage(),
 							Command:         []string{"/usr/local/bin/entrypoint", "/usr/local/bin/webhook-server"},
 							ImagePullPolicy: corev1.PullAlways,
+							LivenessProbe: &corev1.Probe{
+								Handler: corev1.Handler{
+									HTTPGet: &corev1.HTTPGetAction{
+										Path:   "/healthz",
+										Port:   intstr.FromString("liveness-port"),
+										Scheme: "HTTP",
+									},
+								},
+								InitialDelaySeconds: 15,
+								TimeoutSeconds:      5,
+								PeriodSeconds:       10,
+								SuccessThreshold:    1,
+								FailureThreshold:    5,
+							},
+							ReadinessProbe: &corev1.Probe{
+								Handler: corev1.Handler{
+									HTTPGet: &corev1.HTTPGetAction{
+										Path:   "/readyz",
+										Port:   intstr.FromString("liveness-port"),
+										Scheme: "HTTP",
+									},
+								},
+								InitialDelaySeconds: 15,
+								TimeoutSeconds:      5,
+								PeriodSeconds:       10,
+								SuccessThreshold:    1,
+								FailureThreshold:    10,
```

Comment:
If I'm not mistaked, with this settings container will be restart due liveness probe failure before readiness probe fails.

Reply:
Yeah, that's correct, is this okay?

Reply:
We agreed on the call to make that it makes readiness probe useless.

The one I propose:
```
Liveness 
InitialDelaySeconds: 15, #start checking if container needs to be restarted is started in 15 seconds
TimeoutSeconds:      5, 
PeriodSeconds:       10,
SuccessThreshold:    1,  # once one request succeed, it does not need to be restared.
FailureThreshold:    5,  # Restart container if it does not respond 5*10 seconds
```
^ 
```
Readinsess
InitialDelaySeconds: 10, # start checking if container can receive a traffic in 10 seconds.
TimeoutSeconds:      5,
PeriodSeconds:       10,
SuccessThreshold:    1, # once one request succeed, it can receive external traffic
FailureThreshold:    3, # Temporary do not sent traffic to container if it's not available for 3*10 seconds
```

Reply:
Thanks again Serhii! I've updated the code to reflect these changes!

Reply:
Not terribly important, but we generally expect the devworkspace servers to start _fairly_ quickly. The readiness values above mean that either the controller starts 10 seconds after the pod is created, or 20 seconds after the pod is created. Since the readiness probes are only run once per boot, it might make sense to use more frequent checking:
```
Readiness
InitialDelaySeconds: 3
TimeoutSeconds: 3
PeriodSeconds: 3
SuccessThreshold: 1
FailureThreshold: 10 # meaning 10*3 = 30 seconds to start before it's considered failed.
```

It doesn't have a big effect on production deployments, but for development use-cases it might slow things down when re-deploying to only check for readiness 10 seconds after the pod is up.

IMO the first liveness check should happen after the maximum time for readiness checks (30 seconds). With the setup above, we could have liveness probes succeed before the readiness probe (if the readiness probe misses the first window)

Reply:
> Since the readiness probes are only run once per boot

Really??? I need revise my knowledge, probably I misunderstood things. Probes were always confusing stuff

Reply:
from https://v1-18.docs.kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/

> Note: Readiness probes runs on the container during its whole lifecycle.


> Readiness and liveness probes can be used in parallel for the same container. Using both can ensure that traffic does not reach a container that is not ready for it, and that containers are restarted when they fail.

Reply:
But stuff that I did not know before: startup probes. But they seems to be related only to liveness probes(not readiness), it does not seem to help us in the case described by Angel.

I'm OK with the proposed:
```
Readiness
InitialDelaySeconds: 3
TimeoutSeconds: 3
PeriodSeconds: 3
SuccessThreshold: 1
FailureThreshold: 10 # meaning 10*3 = 30 seconds to start before it's considered failed.
```
but not sure how much 3s period is going to generate workload for K8s or controller. Probably little enough that we don't feel difference.

Reply:
Ah maybe I misunderstood readiness probes :upside_down_face:

Reply:
You just prove my statement

> Probes were always confusing stuff

despite their simple idea they tend to be difficult to understand for some reason